server/gaps/gaps_insert_functions.py

```diff
@@ -0,0 +1,167 @@
+
+from helpers import cold_storage_type_id_lookup
+
+
+def changelog_stmt(record_id, table_name):
+    # Thank you to whoever set the default value of cursor in ChangeLog to properly generate the next value!
+    # nextval('changelog_cursor_seq'::regclass)
+    return f"INSERT INTO changelog (record_id, table_name, row_action) VALUES ('{record_id}', '{table_name}', 'UPSERT');"
+
+
+def item_link_stmt(item_id):
+    return f"INSERT INTO item_link (id, item_id) VALUES ('{item_id}', '{item_id}') ON CONFLICT DO NOTHING;"
+
+def upsert_item_stmt(item_id, row):
+    item_name = row['VaccineType']
+    item_code = row['VaccineTypeName']
+    ven_category = "NOT_ASSIGNED"
+    item_type = "STOCK"
+    legacy_record = ""
+    is_active = True
+    is_vaccine = True
+    vaccine_doses = 1 # default, the variants will have the actual dose count
+
+    # Insert the item
+    insert_statement = (f"INSERT INTO item (id, name, code, ven_category, type, legacy_record, is_active, is_vaccine, vaccine_doses, default_pack_size)"
+                        f" VALUES ('{item_id}', '{item_name}', '{item_code}', '{ven_category}', '{item_type}', '{legacy_record}', '{is_active}', '{is_vaccine}', {vaccine_doses}, 1)"
+                        f" ON CONFLICT DO NOTHING;") # For now we just won't update, so only new items will be inserted
+    print(insert_statement)
+    print(item_link_stmt(item_id))
+
+    #Insert the changelog
+    print(changelog_stmt(item_id, 'item'))
+
+    return insert_statement
+
+def upsert_diluent_stmt(item_id, row):
+    item_name = row['VaccineTypeName'] + " Diluent"
+    item_code = row['VaccineTypeName'] + "_DILUENT"
+    ven_category = "NOT_ASSIGNED"
+    item_type = "STOCK"
+    legacy_record = ""
+    is_active = True
+    is_vaccine = False
+    vaccine_doses = 0
+
+    # Insert the item
+    insert_statement = (f"INSERT INTO item (id, name, code, ven_category, type, legacy_record, is_active, is_vaccine, vaccine_doses, default_pack_size)"
+                        f" VALUES ('{item_id}', '{item_name}', '{item_code}', '{ven_category}', '{item_type}', '{legacy_record}', '{is_active}', '{is_vaccine}', {vaccine_doses}, 1)"
+                        f" ON CONFLICT DO NOTHING;") # For now we just won't update, so only new items will be inserted
+    print(insert_statement)
+    print(item_link_stmt(item_id))
+
+    #Insert the changelog
+    print(changelog_stmt(item_id, 'item'))
+
+    return insert_statement
+
+def upsert_item_variant_stmt(item_variant_id, item_id, row):
+    item_variant_name = row['VaccineTypeName'] + " " + row['CommercialName']
+    if row['VaccineTypeName'] == row['CommercialName']:
+        item_variant_name = row['VaccineTypeName']
```

Comment:
NIT, we can improve later... but maybe a bit much duplication 😁 

<img width="427" alt="Screenshot 2024-11-19 at 11 11 14 AM" src="https://github.com/user-attachments/assets/cd559b7e-0128-4c5c-b42b-1e36fffabd6d">